manager.go

```diff
@@ -249,3 +272,49 @@ func (m *Manager) GetRetries(page uint64, pageSize int64, match string) (Retries
 		RetryJobs:       retryJobs,
 	}, nil
 }
+
+func (m *Manager) startHeartbeat() error {
+	if !m.opts.Heartbeat {
+		return nil
+	}
+
+	m.sendHeartbeat()
+
+	heartbeatTicker := time.NewTicker(5 * time.Second)
+	m.heartbeatChannel = make(chan bool, 1)
+
+	go func() {
```

Comment:
Does this need to be another routine?  It looks like the caller of this function invokes a routine as well.